src/main/java/httpserver/httpmessages/HTTPResponseBuilder.java

```diff
@@ -0,0 +1,27 @@
+package httpserver.httpmessages;
+
+import httpserver.ResourceHandler;
+import httpserver.ResponseBuilder;
+import httpserver.resourcemanagement.Resource;
+
+import static httpserver.httpmessages.HTTPResponses.*;
+
+public class HTTPResponseBuilder implements ResponseBuilder {
+
+    private final ResourceHandler resourceHandler;
+
+    public HTTPResponseBuilder(ResourceHandler resourceHandler) {
+        this.resourceHandler = resourceHandler;
+    }
+
+    public HTTPResponse buildResponse(String method, String path) {
+        Resource resource = resourceHandler.getResource(path);
+        if (resource.exists()) {
+            HTTPResponse httpResponse = new HTTPResponse(OK.code, OK.reason);
+            httpResponse.setContentType(resource.getContentType());
+            httpResponse.setBody(resource.getContents());
+            return httpResponse;
+        }
+        return new HTTPResponse(NOTFOUND.code, NOTFOUND.reason);
```

Comment:
I would write something like
``` java
if (resource.exists()) {
    return httpResponse(resource)
} else {
    return new HTTPResponse(NOTFOUND.code, NOTFOUND.reason)
}
```

This way I don't search where are the return, it's pretty obvious and it's also shorter to read